presto-main/src/main/java/com/facebook/presto/metadata/FunctionListBuilder.java

```diff
@@ -87,21 +83,13 @@ public FunctionListBuilder aggregate(Class<?> aggregationDefinition)
 
     public FunctionListBuilder scalar(Class<?> clazz)
     {
-        ScalarFunction scalarAnnotation = clazz.getAnnotation(ScalarFunction.class);
-        ScalarOperator operatorAnnotation = clazz.getAnnotation(ScalarOperator.class);
-        if (scalarAnnotation != null || operatorAnnotation != null) {
-            functions.addAll(ReflectionParametricScalar.parseDefinition(clazz));
-            return this;
-        }
+        functions.addAll(ReflectionParametricScalar.parseFunctionDefinition(clazz));
+        return this;
+    }
 
-        boolean foundOne = false;
-        for (Method method : clazz.getMethods()) {
-            if (ReflectionParametricScalar.canParseMethodDefinition(method)) {
-                functions.addAll(ReflectionParametricScalar.parseDefinition(method));
-                foundOne = true;
-            }
-        }
-        checkArgument(foundOne, "Expected class %s to be annotated with @%s, or contain at least one method annotated with @%s", clazz.getName(), ScalarFunction.class.getSimpleName(), ScalarFunction.class.getSimpleName());
+    public FunctionListBuilder scalars(Class<?> clazz)
```

Comment:
This function is a duplicate of `scalar`, so it should be a delegate (if we have it at all).